apps/wallet-mobile/src/features/Portfolio/common/PortfolioTokenActivityProvider.tsx

```diff
@@ -0,0 +1,101 @@
+import {isNonNullable} from '@yoroi/common'
+import {Portfolio} from '@yoroi/types'
+import React from 'react'
+import {useQuery, useQueryClient} from 'react-query'
+import {merge, switchMap} from 'rxjs'
+
+import {time} from '../../../kernel/constants'
+import {queryInfo} from '../../../kernel/query-client'
+import {useSelectedNetwork} from '../../WalletManager/common/hooks/useSelectedNetwork'
+import {useWalletManager} from '../../WalletManager/context/WalletManagerProvider'
+
+const queryKey = [queryInfo.keyToPersist, 'portfolioTokenActivity']
+const useTokenActivity = () => {
+  const {walletManager} = useWalletManager()
+  const {
+    networkManager: {tokenManager},
+  } = useSelectedNetwork()
+  const [aggregatedBalances, setAggregatedBalances] = React.useState<Portfolio.Token.AmountRecords>()
+  const [tokenIds, setTokenIds] = React.useState<Portfolio.Token.Id[]>()
+  const queryClient = useQueryClient()
+
+  React.useEffect(() => {
```

Comment:
Here's my reasoning.
AggregatedBalances and TokenActivity could be separate, they're independent features.
But they basically use the same subscription, so I think it's a waste of resources to have them both running the same thing when it can be just executed once.
Having this as an extracted hook may give the impression that it can be reused, which there's no need to.